src/Geno/C++/Components/INode.cpp

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) 2021 Sebastian Kylander https://gaztin.com/
+ *
+ * This software is provided 'as-is', without any express or implied warranty. In no event will
+ * the authors be held liable for any damages arising from the use of this software.
+ *
+ * Permission is granted to anyone to use this software for any purpose, including commercial
+ * applications, and to alter it and redistribute it freely, subject to the following restrictions:
+ *
+ * 1. The origin of this software must not be misrepresented; you must not claim that you wrote the
+ *    original software. If you use this software in a product, an acknowledgment in the product
+ *    documentation would be appreciated but is not required.
+ * 2. Altered source versions must be plainly marked as such, and must not be misrepresented as
+ *    being the original software.
+ * 3. This notice may not be removed or altered from any source distribution.
+ */
+
+#include "INode.h"
+
+inline unsigned int NodeIdCounter = 0;
+
+INode::INode( std::filesystem::path Location, std::string Name, NodeKind Kind )
+	: m_Kind( std::move( Kind ) )
+	, m_Name( std::move( Name ) )
+	, m_Location( std::move( Location ) )
```

Comment:
Very nice utilization of move semantics! But `Kind` is an enum (which means it's an integral) and won't have a move constructor, so `std::move( Kind )` here is redundant!